_ml-commons-plugin/cluster-settings.md

```diff
@@ -99,34 +92,63 @@ Controls how many upload model tasks can run in parallel on one node. If set to
 
 ## Load model tasks per node
 
+Controls how many load model tasks can run in parallel on one node. If set as 0, you cannot load models to any node.
+
 ### Setting
 
 ```
 plugins.ml_commons.max_load_model_tasks_per_node: 10
 ```
 
-### Description
-
-Controls how many load model tasks can run in parallel on one node. If set as 0, you cannot load models to any node.
-
 ### Values 
 
 - Default value: `10`
 - Value range: [0, 10]
 
 ## Add trusted URL
 
+The default value allows uploading a model file from any http/https/ftp/local file. You can change this value to restrict trusted model URL.
+
+
 ### Setting
 
 ```
 plugins.ml_commons.trusted_url_regex: ^(https?\|ftp\|file)://[-a-zA-Z0-9+&@#/%?=~_\|!:,.;]*[-a-zA-Z0-9+&@#/%=~_\|]
 ```
 
-### Description
-
-The default value allows uploading a model file from any http/https/ftp/local file. You can change this value to restrict trusted model URL
-
 ### Values
 
 - Default value: `^(https?\|ftp\|file)://[-a-zA-Z0-9+&@#/%?=~_\|!:,.;]*[-a-zA-Z0-9+&@#/%=~_\|]`
 - Value range: Java regular expression (regex) string
+
+## Assign task timeout
+
+Assigns how long an ML task will live in seconds. After timeout, the task will fail.
+
+### Setting
+
+```
+plugins.ml_commons.ml_task_timeout_in_seconds: 600
+```
+
+### Values
+
+- Default value: 600
+- Value range: [1, 86400]
+
+## Set native memory threshold 
+
+Sets a circuit breaker that checks all system memory usage before running an ML task. If the native memory exceeds the threshold, OpenSearch puts up an exception and stops running any ML task. 
```

Comment:
```suggestion
Sets a circuit breaker that checks all system memory usage before running an ML task. If the native memory exceeds the threshold, OpenSearch throws an exception and stops running any ML task. 
```

Reply:
I think we mean "throws" here, but if not, please replace the phrasal verb "puts up" with something more descriptive.